conn.go

```diff
@@ -87,7 +87,7 @@ func createConn(ctx context.Context, nextConn net.Conn, config *Config, isClient
 		return nil, errNilNextConn
 	}
 
-	cipherSuites, err := parseCipherSuites(config.CipherSuites, config.PSK == nil || len(config.Certificates) > 0, config.PSK != nil)
+	cipherSuites, err := parseCipherSuites(config.CipherSuites, config.CustomCipherSuites, config.PSK == nil || len(config.Certificates) > 0, config.PSK != nil)
```

Comment:
Do we want to rename `config.PSK` here too?

Reply:
Sorry I think I am not understanding the feedback!

`config.CustomCipherSuites` is a new member. I am happy to rename `config.PSK` and we can deprecate/support both until `/v3` if you think that is best.

Reply:
Mostly just that. I say you introduced AuthenticationType, which I find more clear/expressive than PSK. So just wondering if we should go about and remove PSK as nomenclature entirely. But yes, can do that later.